src/libraries/System.Runtime/tests/System/Runtime/JitInfoTests.cs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.XUnitExtensions;
+using System.Threading;
+using Xunit;
+
+namespace System.Runtime.Tests
+{
+    public class JitInfoTests
+    {
+        private const TestPlatforms AotPlatforms = TestPlatforms.Browser | TestPlatforms.iOS | TestPlatforms.tvOS | TestPlatforms.Android;
+
+        [Fact]
+        [SkipOnPlatform(AotPlatforms, "JitInfo metrics will be 0 in AOT scenarios.")]
+        public void JitInfoIsPopulated()
+        {
+            TimeSpan beforeCompilationTime = System.Runtime.JitInfo.GetCompilationTime();
+            long beforeCompiledILBytes = System.Runtime.JitInfo.GetCompiledILBytes();
+            long beforeCompiledMethodCount = System.Runtime.JitInfo.GetCompiledMethodCount();
+
+            Func<string> theFunc = () => "JIT compile this!";
+            Assert.True(theFunc().Equals("JIT compile this!"));
```

Comment:
This will fail with test runs that use R2R. The method will be precompiled in that case and there will be no JITing.

You may want to dynamically emit a method instead. Dynamically emitted methods cannot be ever AOT compiled by design.